src/plugins/query_enhancements/common/types.test.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { JobState, parseJobState } from './types';
+
+describe('parseJobState', () => {
+  it('should return the correct JobState for a valid string', () => {
+    expect(parseJobState('SUBMITTED')).toBe(JobState.SUBMITTED);
+    expect(parseJobState('waiting')).toBe(JobState.WAITING);
+    expect(parseJobState('SCHEDULED')).toBe(JobState.SCHEDULED);
+    expect(parseJobState('running')).toBe(JobState.RUNNING);
+    expect(parseJobState('FAILED')).toBe(JobState.FAILED);
+    expect(parseJobState('success')).toBe(JobState.SUCCESS);
+    expect(parseJobState('CANCELLING')).toBe(JobState.CANCELLING);
+    expect(parseJobState('cancelled')).toBe(JobState.CANCELLED);
```

Comment:
nit: why do we have different cases for the states, some are capitalized while others arent

Reply:
Testing case-insensitivity, though there's probably a way to make it more obvious